### PR TITLE
CTO-118 Updated Gate Change Controller and DAO

### PR DESCRIPTION
**Pull request by zack-rma**
Updated Gate Change Controller and DAO to support operation changes store/delete functionality

### CODE REVIEW

File: cwms-data-api/src/main/java/cwms/cda/data/dao/location/kind/OutletDao.java [hunk 1]
```diff
@@ -114,8 +117,12 @@ public void storeOutlet(Outlet outlet, boolean failIfExists) {
     public void deleteOutlet(String officeId, String locationId, DeleteRule deleteRule) {
         connection(dsl, conn -> {
             setOffice(conn, officeId);
-            CWMS_OUTLET_PACKAGE.call_DELETE_OUTLET(DSL.using(conn).configuration(), locationId, deleteRule.getRule(),
-                                                   officeId);
+            try {
+                CWMS_OUTLET_PACKAGE.call_DELETE_OUTLET(DSL.using(conn).configuration(), locationId,
+                        deleteRule.getRule(), officeId);
+            } catch (IntegrityConstraintViolationException e) {
+                throw new NotFoundException(e);
```
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/data/dao/location/kind/OutletDao.java]: Not found doesn't make sense given the parent exception. `409 Conflict`  https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409 makes more sense.

File: cwms-data-api/src/test/java/cwms/cda/data/dto/location/kind/GateChangeTest.java [hunk 2]
```diff
@@ -199,6 +199,7 @@ static GateChange buildTestGateChange(CwmsId projectId, Instant changeDate, Gate
                                        .withNotes(notes)
                                        .withChangeDate(changeDate)
                                        .withSettings(settings)
+//                                       .withType("gate-change")
```
Comment (MikeNeilson) [inline cwms-data-api/src/test/java/cwms/cda/data/dto/location/kind/GateChangeTest.java]: Either delete if not needed, or provide explanation of why it's sitting here commented out (in the comment itself.)

File: cwms-data-api/src/test/resources/cwms/cda/api/gate_change.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+[ {
+  "type" : "gate-change",
+  "project-id" : {
+    "office-id" : "SPK",
+    "name" : "PROJECT1"
```
Comment (MikeNeilson) [inline cwms-data-api/src/test/resources/cwms/cda/api/gate_change.json]: use a "real" project name.

File: cwms-data-api/src/test/resources/cwms/cda/api/gate_change.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  "new-total-discharge-override" : 1.0,
+  "old-total-discharge-override" : 2.0,
+  "discharge-units" : "cfs",
+  "tailwater-elevation" : 4.0,
+  "elevation-units" : "ft",
+  "settings" : [ {
+    "type" : "gate-setting",
+    "location-id" : {
+      "office-id" : "SPK",
+      "name" : "PROJECT1-CG100"
```
Comment (MikeNeilson) [inline cwms-data-api/src/test/resources/cwms/cda/api/gate_change.json]: same as above comment, use a "real" name.

**Review by MikeNeilson — CHANGES_REQUESTED** [review]
Generally looks good. Some changes for clarity required.

File: cwms-data-api/src/test/resources/cwms/cda/api/gate_change.json (hunk 4 above)
Comment (zack-rma) [reply cwms-data-api/src/test/resources/cwms/cda/api/gate_change.json]: Swapped for a project name used in other Outlet testing

File: cwms-data-api/src/test/resources/cwms/cda/api/gate_change.json (hunk 3 above)
Comment (zack-rma) [reply cwms-data-api/src/test/resources/cwms/cda/api/gate_change.json]: Swapped for a project name used in other Outlet testing

File: cwms-data-api/src/test/java/cwms/cda/data/dto/location/kind/GateChangeTest.java (hunk 2 above)
Comment (zack-rma) [reply cwms-data-api/src/test/java/cwms/cda/data/dto/location/kind/GateChangeTest.java]: Deleted unnecessary comment

File: cwms-data-api/src/main/java/cwms/cda/data/dao/location/kind/OutletDao.java (hunk 1 above)
Comment (zack-rma) [reply cwms-data-api/src/main/java/cwms/cda/data/dao/location/kind/OutletDao.java]: Swapped for a DeleteConflictException. The error code received by HCDA isn't the most useful, ("Data already exists for this request"). Maybe there's another exception that will provide more detail?

Comment (MikeNeilson) [reply cwms-data-api/src/main/java/cwms/cda/data/dao/location/kind/OutletDao.java]: It should send basically what the constraint is "something depends on it." We'll have to look into that later.

**Review by MikeNeilson — APPROVED** [review]

**Comment by MikeNeilson** [comment]
There are failing tests, but they look unrelated to this PR so I'm merging in anyways.

**Pull request merged**